magefiles/docs.go

```diff
@@ -26,4 +39,131 @@ func main() {
+	remoteFlags := flag.NewClientFlags()
+	remoteFlags.Listen = flag.ServerListenFlag.Clone()
+
+	// These flags don't work from config file.
+	// Clear configName to skip them later.
+	globalFlags := flag.NewGlobalFlagGroup()
+	globalFlags.ConfigFile.ConfigName = ""
+	globalFlags.ShowVersion.ConfigName = ""
+	globalFlags.GenerateDefaultConfig.ConfigName = ""
+
+	var allFlagGroups = []flag.FlagGroup{
+		globalFlags,
+		remoteFlags,
+		flag.NewCacheFlagGroup(),
+		flag.NewCleanFlagGroup(),
+		flag.NewDBFlagGroup(),
+		flag.NewImageFlagGroup(),
+		flag.NewK8sFlagGroup(),
+		flag.NewLicenseFlagGroup(),
+		flag.NewMisconfFlagGroup(),
+		flag.NewModuleFlagGroup(),
+		flag.NewRegistryFlagGroup(),
+		flag.NewRegoFlagGroup(),
+		flag.NewReportFlagGroup(),
+		flag.NewRepoFlagGroup(),
+		flag.NewScanFlagGroup(),
+		flag.NewSecretFlagGroup(),
+		flag.NewVulnerabilityFlagGroup(),
+		flag.NewAWSFlagGroup(),
+	}
+
+	f, err := os.Create(filename)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+	f.WriteString("# " + title + "\n\n")
+	f.WriteString(description + "\n")
+
+	for _, group := range allFlagGroups {
+		f.WriteString("## " + group.Name() + " options\n")
+		writeFlags(group, f)
+	}
+
+	f.WriteString(footer)
+	return nil
+}
+
+func writeFlags(group flag.FlagGroup, w *os.File) {
+	flags := group.Flags()
+	slices.SortFunc(flags, func(a, b flag.Flagger) int {
+		return cmp.Compare(a.GetConfigName(), b.GetConfigName())
+	})
+	w.WriteString("\n```yaml\n")
+
+	var lastParts []string
+	for _, flg := range flags {
+		if flg.GetConfigName() == "" {
+			continue
+		}
+		parts := strings.Split(flg.GetConfigName(), ".")
+		for i := range parts {
+			// Skip already added part
+			if len(lastParts) >= i+1 && parts[i] == lastParts[i] {
+				continue
+			}
+			ind := strings.Repeat("  ", i)
+			isLastPart := i == len(parts)-1
+			if isLastPart {
+				if flg.GetName() != "" {
+					fmt.Fprintf(w, "%s# Same as '--%s'\n", ind, flg.GetName())
+				}
+				fmt.Fprintf(w, "%s# Default is %v\n", ind, defaultValueString(flg.GetDefaultValue()))
+			}
+			w.WriteString(ind + parts[i] + ":")
+			if isLastPart {
+				writeFlagValue(flg.GetDefaultValue(), ind, w)
+			}
+			w.WriteString("\n")
+		}
+		lastParts = parts
+	}
+	w.WriteString("```\n")
+}
+
+func defaultValueString(val any) string {
+	var value string
+	switch v := val.(type) {
+	case string:
+		value = lo.Ternary(len(v) > 0, v, "empty")
+	case []string:
+		value = lo.Ternary(len(v) > 0, strings.Join(v, ", "), "[]")
+	default:
+		value = fmt.Sprintf("%v", v)
+	}
+	return value
+}
+
+func writeFlagValue(val any, ind string, w *os.File) {
+	switch v := val.(type) {
+	case []string:
+		if len(v) > 0 {
+			w.WriteString("\n")
+			for _, vv := range v {
+				fmt.Fprintf(w, "%s - %s\n", ind, vv)
+			}
+		} else {
+			w.WriteString(" empty\n")
+		}
+	case string:
+		if len(v) > 0 {
+			fmt.Fprintf(w, " %v\n", v)
+		} else {
+			w.WriteString("\n")
+		}
+	default:
+		fmt.Fprintf(w, " %v\n", v)
+	}
```

Comment:
We can remove case for `string`.
`default` also works correctly for `string`
We get extra space after `:` for empty strings, but this is still valid `yaml`.
```suggestion
	default:
		fmt.Fprintf(w, " %v\n", v)
	}
```

Reply:
ok, done